crates/matrix-sdk-ui/src/timeline/pagination.rs

```diff
@@ -163,22 +132,20 @@ impl super::Timeline {
                     }
                 }
 
-                // Retrieve the next earliest back-pagination token.
-                token = self.event_cache.oldest_backpagination_token(None).await?;
-
                 // Exit the inner loop, and ask for another limit.
                 break;
             }
         }
 
-        // The status is automatically reset to idle by `reset_status_guard`.
-
         Ok(false)
     }
 
     /// Subscribe to the back-pagination status of the timeline.
-    pub fn back_pagination_status(&self) -> Subscriber<PaginationStatus> {
-        self.back_pagination_status.subscribe()
+    ///
+    /// Note: this may send multiple Paginating/Idle sequences during a single
```

Comment:
Why? Do we want to dedup them? Maybe with [`async_rx::StreamExt::dedup`](https://docs.rs/async-rx/latest/async_rx/trait.StreamExt.html#method.dedup) if we can provide a `Stream`.

Reply:
It's an implementation detail of `eyeball` that we're running into. If you have an emitter E and an observer O. Say E emits very quickly e1, e2, e3; then if O doesn't react fast enough, it may only see e1, handle it, and then miss e2, and only see e3.

Before this PR, the exposed pagination status was controlled at the timeline level, and the timeline would make sure to reset it only when a full pagination loop is over, after we obtained e.g. N events.

After this PR, the pagination status is that of the underlying paginator. The timeline may continue to paginate after the paginator has run one pagination query. In that case, the underlying status of the paginator will be:

- running (because it's busy doing the network request)
- idle (because it's done processing the response)
- (the timeline sees it hasn't received as many events as it wanted, so it retriggers the paginator) the paginator quickly re-runs into the running state

So the quick `running -> idle -> running` transition might make it so that an observer could see `running -> running`, because they didn't react fast enough to see the `idle` state in the middle.

I've added a `dedup()` call to address this, thanks.

Now, if the caller *is* actually fast enough, they may see a few quick transitions from `running -> idle -> running`; I wonder if that may cause some UI flickering (e.g. the spinner quickly disappears and reappears). At this point, both EX apps doesn't use the back-pagination status anymore, so it's a theoretical issue.